pkg/registry/apis/query/parser.go

```diff
@@ -189,6 +181,19 @@ func (p *queryParser) parseRequest(ctx context.Context, input *query.QueryDataRe
 	return rsp, nil
 }
 
+func getTimeRangeForQuery(parentTimerange, queryTimerange *data.TimeRange) data.TimeRange {
+	if queryTimerange != nil && queryTimerange.From != "" && queryTimerange.To != "" {
+		return *queryTimerange
+	}
+	if parentTimerange != nil && parentTimerange.To != "" && parentTimerange.From != "" {
+		return *parentTimerange
+	}
+	return data.TimeRange{
```

Comment:
is it valid to make a request without a time range currently? i would expect an error

Reply:
Without any is also a bit strange to me, I guess that's something we could change as part of a new API version

Reply:
missing values are currently sent on as zero:
https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/gtime/time_range.go#L64

Reply:
is `now-6h` coming from existing code? to me it makes more sense to error or to leave it empty if the query and parent request don't have a time range set. setting it to `now-6h` might be a good default for some data sources, or result in an expensive query for others

Reply:
@JohnnyQQQQ -- can you update the PR so missing time gets a range 0-0 like it did before?

Reply:
What would 0-0 be in terms or timerange string? `now`, `now`?

Reply:
I guess I can set them to `0` and `0`.

Reply:
i think leaving them empty would also do it

Reply:
Just tested that locally so it should be fine

```
		val := gtime.NewTimeRange("0", "0")
		tr := &backend.TimeRange{
			From: val.GetFromAsTimeUTC(),
			To:   val.GetToAsTimeUTC(),
		}
		require.Equal(t, tr.Duration(), time.Duration(0))
```